### DISCUSSION TITLE
opencsg如何设置casdoor为身份验证和授权提供商

### DISCUSSION THREAD

**Discussion by miss-uny** [OpenCSGs/csghub#597] [Q&A]
请问opencsg如何设置casdoor为身份验证和授权提供商
我想达到的目标是 通过casdoor登录后，可以直接转入csghub 。 现在已经实现了设置好了casdoor，可以登录，但无法跳转到csghub。
查询相关的文档也未见相关说明，请提供配置方法，感谢

**Comment by hiveer** [comment]
我们通过一个环境变量控制登录方式的接入，这个环境变量是： ON_PREMISE

1. 如果 ON_PREMISE=true 那么系统将采用自带的一个简易登录方式
2. 如果 ON_PREMISE=false 那么这个时候就可以通过支持 OIDC 的登录方式登录，比如 casdoor

你需要添加如下的环境变量：
OIDC_IDENTIFIER=
OIDC_SECRET=
OIDC_REDIRECT_URI=http://<csghub-domain>/oidc/callback
OIDC_AUTHORIZATION_ENDPOINT=
OIDC_TOKEN_ENDPOINT=
OIDC_USERINFO_ENDPOINT=

**Comment by wayneliu0019** [comment]
这里给你一个配置示例：

假设用户部署的csghub实例的url为`http://aihub.mycorp.cn`, 启动的casdoor实例url为`http://aihub.mycorp.cn:8000`，那么portal服务需要更改的配置项目如下： 

```
      ON_PREMISE: false
      OIDC_IDENTIFIER: 57d8874xxxxxxxxxf97
      OIDC_SECRET: 88790798803xxxxxxxxxx5aaa5a5bd8b44d6922d93
      OIDC_REDIRECT_URI: http://aihub.mycorp.cn/oidc/callback
      OIDC_AUTHORIZATION_ENDPOINT: http://aihub.mycorp.cn:8000/login/oauth/authorize
      OIDC_TOKEN_ENDPOINT: http://aihub.mycorp.cn:8000/api/login/oauth/access_token
      OIDC_USERINFO_ENDPOINT: http://aihub.mycorp.cn:8000/api/userinfo
      LOGIN_URL: http://aihub.mycorp.cn:8000/login/oauth/authorize?client_id=57d8874xxxxxxxxxf97&response_type=code&redirect_uri=http://aihub.mycorp.cn/oidc/callback&scope=read&state=casdoor
      SIGNUP_URL: http://aihub.mycorp.cn:8000/signup/portal
```

上面`OIDC_IDENTIFIER`和`OIDC_SECRET`分别是在casdoor中创建的appliation对应的client_id和client_secret, 截图如下：
<img width="840" alt="image" src="https://github.com/user-attachments/assets/26c71b98-9a4d-44ed-b2aa-e32386759aad">
